pkg/apic/servicebuilder.go

```diff
@@ -25,6 +26,7 @@ type ServiceBuilder interface {
 	SetState(state string) ServiceBuilder
 	SetStatus(status string) ServiceBuilder
 	SetServiceAttribute(serviceAttribute map[string]string) ServiceBuilder
+	SetServiceEndpoints(endpoints []v1alpha1.ApiServiceInstanceSpecEndpoint) ServiceBuilder
```

Comment:
```suggestion
	AddServiceEndpoint(protocol, host string, port int, basePath string) ServiceBuilder
```

Reply:
@tsjohns9, as we discussed can we instead of  having []v1alpha1.ApiServiceInstanceSpecEndpoint for building endpoints have a array of struct (say EndpointDefinition or something) holding the properties in ServiceBody and in ServiceBodyBuilder.AddServiceEndpoint setup the struct and append to array in ServiceBody. 

While processing the publishing in proessInstance() method pkg/apic/apiserviceInstance.go we can prepare the []v1alpha1.ApiServiceInstanceSpecEndpoint using the array of EndpointDefinition in ServiceBody.